src/avram/type.cr

```diff
@@ -11,6 +11,18 @@ module Avram::Type
     SuccessfulCast(Nil).new(nil)
   end
 
+  def parse(values : Array(String))
+    values = values.map do |value|
+      parsed_result = parse(value)
+      if parsed_result.is_a? Avram::Type::SuccessfulCast
+        parsed_result.value
+      else
+        nil
+      end
```

Comment:
This is due to that if a parse fails, `FailedCast` doesn't have a `value` method. Maybe if even one fails, the whole thing should? How would that look though?

Reply:
I think if one fails, they all should. Not sure how it would look though. Will need to dig in more later!